runhouse/resources/hardware/cluster.py

```diff
@@ -385,6 +385,18 @@ def check_server(self, restart_server=True):
                 logger.info(f"Checking server {self.name}")
                 self.client.check_server(cluster_config=cluster_config)
                 logger.info(f"Server {self.name} is up.")
+
+                import runhouse
```

Comment:
ideas for checking version are `import runhouse; runhouse.__version__` and `pip freeze | grep runhouse`, I think the first is faster?

a bit concerned on latency though, since `check_server` is called quite frequently (most cluster calls will pass through here), does it make sense to just call this when we instantiate a cluster and if it's already up, then check?

Reply:
What if we run the check whenever the HTTP server is started / restarted (in the constructor)? Or could the mismatch still exist in other spots that we would miss?

Reply:
from discussion today: could consider caching the runhouse version as a global variable on remote server when server is started, and compare local runhouse version with that. local import and version calls should be quick if runhouse has already been imported.